### PR TITLE
Add multiple algorithm support in compute_igraph

### PR DESCRIPTION
**Pull request by sky-2002**
@silkspace , This PR tries to solve issue mentioned [here](https://github.com/graphistry/pygraphistry/issues/405). With this, multiple algorithms can be passed as a list to the `compute_igraph` function.

### CODE REVIEW

**Comment by lmeyerov** [comment]
@sky-2002  see my comment on https://github.com/graphistry/pygraphistry/issues/405 wrt more usable signature

looks like lint/type errors, see ci . you can build & test locally

see https://github.com/graphistry/pygraphistry/blob/master/DEVELOP.md#run-local-tests-without-rebuild + look at the script for env vars you can set

**Comment by sky-2002** [comment]
Hello @lmeyerov , @silkspace , I have made the necessary changes for more usable signature, added code to support custom fieldnames as suggested by @DataBoyTX in #405 and added tests too. Let me know if anything needs to be changed.

File: graphistry/tests/plugins/test_igraph.py [hunk 1]
```diff
@@ -516,6 +516,16 @@ def test_all_calls(self):
                         warnings.filterwarnings("ignore", category=FutureWarning)
                         assert compute_igraph(g, alg, **opts) is not None
 
+        multiple1 = compute_igraph(g,{'pr':('pagerank',{'damping': 0.85}),
+                                     'coci':'cocitation','cl':'closeness','bt':'betweenness'})
+        assert 'cl' in multiple1._nodes.columns
+        assert 'bt' in multiple1._nodes.columns
```
Comment (lmeyerov) [inline graphistry/tests/plugins/test_igraph.py:522]: assert also `pr` & `coci`

File: graphistry/tests/plugins/test_igraph.py [hunk 2]
```diff
@@ -516,6 +516,16 @@ def test_all_calls(self):
...
+
+        multiple2 = compute_igraph(g, [('pagerank',{'damping': 0.85}),'closeness'])
+        assert 'closeness' in multiple2._nodes.columns
```
Comment (lmeyerov) [inline graphistry/tests/plugins/test_igraph.py:525]: assert also `pagerank`

File: graphistry/plugins/igraph.py [hunk 3]
```diff
@@ -282,7 +282,8 @@ def to_igraph(self: Plottable,
 
 def compute_igraph(
     self: Plottable,
-    alg: str,
+    # alg: str,
```
Comment (lmeyerov) [inline graphistry/plugins/igraph.py:285]: remove

File: graphistry/plugins/igraph.py [hunk 4]
```diff
@@ -282,7 +282,8 @@ def to_igraph(self: Plottable,
...
+    alg: Union[str, List[Union[str, Tuple[str, dict]]], Dict[str, Union[str, Tuple[str,dict]]] ],
```
Comment (lmeyerov) [inline graphistry/plugins/igraph.py:286]: Add `** Example...` of the new forms

**Review by lmeyerov — CHANGES_REQUESTED** [review]
Getting there!

* See CI errors
* Can you add an example to `readme.md` of the new forms?
* Can you add to the changelog? 
* See inline review comments

**Comment by sky-2002** [comment]
> Getting there!
> 
> * See CI errors
> * Can you add an example to `readme.md` of the new forms?
> * Can you add to the changelog?
> * See inline review comments

For the changelog, will this be fine - "Multiple algorithm support for igraph" in the [Development] Added section.

**Comment by lmeyerov** [comment]
Looks like the type checker needs help?

**Comment by sky-2002** [comment]
> Looks like the type checker needs help?

Well, tried to fix it, let's see.

**Comment by lmeyerov** [comment]
Great. Did you try running the tests locally? Set the various env vars of interest to `=1` @ https://github.com/graphistry/pygraphistry/blob/master/DEVELOP.md#run-local-tests-without-rebuild

**Comment by lmeyerov** [comment]
We can revisit this when more cycles, closing for now

**Pull request closed without merging**